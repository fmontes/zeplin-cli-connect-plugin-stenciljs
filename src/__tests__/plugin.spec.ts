import Plugin from "../index";

describe("Connected Components StencilJS Plugin - TypeScript", () => {
    let plugin: Plugin;

    describe("config file", () => {
        beforeEach(() => {
            plugin = new Plugin();
        });
    
        test("it should throw error when no source file passed", async () => {
            expect(plugin.init({})).rejects.toEqual(
                new Error(
                    "You need to pass a valid document json file. More information in: https://github.com/fmontes/zeplin-cli-connect-plugin-stenciljs"
                )
            );
        });

        test("it should throw error when no source file can't be parsed", async () => {
            expect(
                plugin.init({
                    config: {
                        sourcePath: "path-dont-work",
                    },
                })
            ).rejects.toEqual(
                new Error(
                    "Can't parse your documentation json file. More information in: https://github.com/fmontes/zeplin-cli-connect-plugin-stenciljs"
                )
            );
        });
    });

    describe("proccess", () => {
        beforeEach(() => {
            plugin = new Plugin();
            plugin.init({
                config: {
                    sourcePath: "./src/__tests__/source.mock.json"
                }
            })
        });
        it("my-compontent.tsx snippet creation", async () => {
            const componentCode = await plugin.process(
                {
                    path: "/my-compontent.tsx",
                    zeplinNames: []
                }
            );
    
            expect(componentCode).toMatchSnapshot();
        })  
    })
});
