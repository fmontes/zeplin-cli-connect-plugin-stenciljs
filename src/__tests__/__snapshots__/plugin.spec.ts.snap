// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Components StencilJS Plugin - TypeScript proccess my-compontent.tsx snippet creation 1`] = `
Object {
  "description": "This is the documentation

Last update: 2020-09-20T05:48:43",
  "lang": "html",
  "snippet": "<my-component
  name={string} />",
}
`;
